### PR TITLE
Add ke17z basic support

### PR DESCRIPTION
**Pull request by ankeXiao**

### CODE REVIEW

**Comment by hakehuang** [comment]
@ankeXiao , there are several build failure

<html>
<body>

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
mgmt.mcumgr.fs.mgmt.hash.supported.crc32 | v3.6.0-3542-g8c0bff829db0 | Build failure | frdm_ke17z | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.sha256 | v3.6.0-3542-g8c0bff829db0 | Build failure | frdm_ke17z | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.all | v3.6.0-3542-g8c0bff829db0 | Build failure | frdm_ke17z | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
sample.drivers.flash.shell | v3.6.0-3542-g8c0bff829db0 | Build failure | frdm_ke17z | samples/drivers/flash_shell


</body>
</html>
`

**Comment by hakehuang** [comment]
now build and board testing are clean.

**Comment by ankeXiao** [comment]
The soc module files depends on this PR "[pull/71670](https://github.com/zephyrproject-rtos/zephyr/pull/71670)" and is shared with KE15Z platform, so will be updated after it is reviewed and merged.

File: boards/nxp/frdm_ke17z/doc/frdmke17z.png
Comment (dleach02) [inline boards/nxp/frdm_ke17z/doc/frdmke17z.png:1]: as commented on your other PR, convert to webp and put through [tinypng.com](https://tinypng.com/)

**Comment by dleach02** [comment]
Please address the compliance error

**Comment by ankeXiao** [comment]
Hi all, there is a compilation failure in the "build-docs" under the "Documentation Build(HTML)" step, I have no idea which file  caused it, please help to check. Thanks!

**Comment by dleach02** [comment]
There was a site-wide CI failure going on. Appears to have been addressed.

File: boards/nxp/frdm_ke17z/frdm_ke17z_defconfig [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+CONFIG_SERIAL=y
+
+# GPIO Controller
+CONFIG_GPIO=y
+
+# Clock Control
+CONFIG_CLOCK_CONTROL=y
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=72000000
```
Comment (danieldegrasse) [inline boards/nxp/frdm_ke17z/frdm_ke17z_defconfig:17]: Is this accurate? The documentation states the board is clocked at 48MHz, and this value should be equal to the clock frequency the SYSTICK is running at on boards using it (which this M0+ will by default). A good way to check if this value is correct is to use a sample like `samples/basic/blinky`, and make sure the LED toggles once per second using an oscilloscope or signal analyzer

Comment (ankeXiao) [reply boards/nxp/frdm_ke17z/frdm_ke17z_defconfig]: Thanks for reminding, the actual frequency should be 48000000, which should be consistent with the index.rsy document.

File: boards/nxp/frdm_ke17z/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+The default configuration can be found in the defconfig file:
+:zephyr_file:`boards/nxp/frdm_ke17z/frdm_ke17z_defconfig`.
+
+Other hardware features are not currently supported by the port.
+
+System Clock
+============
+
+The KE17Z SoC is configured to run at 48 MHz using the FIRC.
```
[the remark this answers is not in the document]
Comment (ankeXiao) [reply boards/nxp/frdm_ke17z/doc/index.rst:71]: Corrected the value of SYS_CLOCK_HW_CYCLES_PER_SEC to 48MHZ.

File: dts/arm/nxp/nxp_ke17z.dtsi
Comment (danieldegrasse) [inline dts/arm/nxp/nxp_ke17z.dtsi:1]: How much of this file duplicates the `ke1xz.dtsi` file? From what I can tell (just glancing, I may be missing differences), we are duplicating the following nodes:
- cpu
- pinctrl
- pcc
- lpuart0-2
- porta-e
- gpioa-e

So we could import the `ke1xz.dtsi` file here, and override/define the following nodes:
- sram_l
- sram_u
- scg
- ftfa


Does this seem right?

Comment (ankeXiao) [reply dts/arm/nxp/nxp_ke17z.dtsi:1]: Only sram and ftfe currently, the scg can be consistent.

**Review by danieldegrasse — DISMISSED** [review]

File: dts/arm/nxp/nxp_ke17z512.dtsi [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+			#size-cells = <1>;
+
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				reg = <0 DT_SIZE_K(512)>;
+				erase-block-size = <DT_SIZE_K(2)>;
+				write-block-size = <8>;
+			};
+		};
+    };
```
Comment (dleach02) [inline dts/arm/nxp/nxp_ke17z512.dtsi:46]: tab instead of space

**Review by dleach02 — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

File: dts/arm/nxp/nxp_ke17z512.dtsi (hunk 3 above)
Comment (ankeXiao) [reply dts/arm/nxp/nxp_ke17z512.dtsi]: Thanks!

**Pull request merged**